tests/MMS/spatial/finite-volume/fv_mms.cxx

```diff
@@ -42,6 +42,15 @@ int main(int argc, char** argv) {
   Field3D v{FieldFactory::get()->create3D("v", Options::getRoot(), mesh)};
 
   // Communicate to calculate parallel transform.
+  if constexpr (bout::build::use_metric_3d) {
+    // Div_par operators require B parallel slices:
+    // Coordinates::geometry doesn't ensure this (yet)
+    auto& Bxy = mesh->getCoordinates()->Bxy;
+    auto& J = mesh->getCoordinates()->J;
```

Comment:
Note that communicating J is likely wrong for any realistic grid. You need #3197 
For a simple slab test, that may well work, but not for real grids.

Reply:
The parallel slices can have a completely different geometry. Thus one needs J from the traced parallel slices. The communication interpolates J, which works for lab-based quantities, but not for mesh based quantities.

One solution is to also write out the parallel metric components in zoidberg, and read them in in BOUT++, and prevent them from being overwritten:
https://github.com/boutproject/zoidberg/blob/more-6/zoidberg/zoidberg.py#L421
https://github.com/boutproject/BOUT-dev/blob/f3dwy/src/mesh/parallel/fci.cxx#L134

See also:
https://bout.zulipchat.com/#narrow/channel/535664-BSTING/topic/Connection.20between.20J.20and.20B.20in.20FCI/with/554530388